presto-orc/src/main/java/com/facebook/presto/orc/StripeReader.java

```diff
@@ -432,18 +445,24 @@ static boolean isIndexStream(Stream stream)
         return selectedRowGroups.build();
     }
 
-    private static Map<Integer, ColumnStatistics> getRowGroupStatistics(OrcType rootStructType, Map<Integer, List<RowGroupIndex>> columnIndexes, int rowGroup)
+    private static Map<Integer, ColumnStatistics> getRowGroupStatistics(OrcType rootStructType, Map<StreamId, List<RowGroupIndex>> columnIndexes, int rowGroup)
     {
         requireNonNull(rootStructType, "rootStructType is null");
         checkArgument(rootStructType.getOrcTypeKind() == OrcTypeKind.STRUCT);
         requireNonNull(columnIndexes, "columnIndexes is null");
         checkArgument(rowGroup >= 0, "rowGroup is negative");
 
+        Map<Integer, List<ColumnStatistics>> columnStatisticsByColumn = new HashMap<>();
+        for (Entry<StreamId, List<RowGroupIndex>> entry : columnIndexes.entrySet()) {
+            columnStatisticsByColumn.computeIfAbsent(entry.getKey().getColumn(), (key) -> new ArrayList<>())
+                    .add(entry.getValue().get(rowGroup).getColumnStatistics());
+        }
+
         ImmutableMap.Builder<Integer, ColumnStatistics> statistics = ImmutableMap.builder();
         for (int ordinal = 0; ordinal < rootStructType.getFieldCount(); ordinal++) {
-            List<RowGroupIndex> rowGroupIndexes = columnIndexes.get(rootStructType.getFieldTypeIndex(ordinal));
-            if (rowGroupIndexes != null) {
-                statistics.put(ordinal, rowGroupIndexes.get(rowGroup).getColumnStatistics());
+            List<ColumnStatistics> columnStatistics = columnStatisticsByColumn.get(rootStructType.getFieldTypeIndex(ordinal));
+            if (columnStatistics != null) {
+                statistics.put(ordinal, ColumnStatistics.mergeColumnStatistics(columnStatistics));
```

Comment:
Yes, at least for now, this is so that Presto can still operate with the column as it does with other maps.  In the future, Presto could be optimized to take advantage of the per value column statistics.